cmd/thanos/compact.go

```diff
@@ -40,6 +43,9 @@ func registerCompact(m map[string]setupFunc, app *kingpin.Application, name stri
 	syncDelay := cmd.Flag("sync-delay", "Minimum age of fresh (non-compacted) blocks before they are being processed.").
 		Default("30m").Duration()
 
+	var retention model.Duration
+	cmd.Flag("storage.retention", "How long to retain samples in storage. 0d - disables retention").Default("0d").SetValue(&retention)
```

Comment:
Can we keep it consistent and return value instead of setting? And why you move it from time.Duration to model.Duration and back to time.Duration?

Reply:
Model Duration supports days, weeks, months, while https://golang.org/pkg/time/#ParseDuration up to an hour, so that's why I did it this way.

Reply:
yes, but is it consistent with other places we do duration? I don't think so. If we want to change this, let's add helper for this (something that will return us a timeDuration properly parsed or just model.Duration) and use it everywhere we do `Duration()`?